### PR TITLE
Introduce nested property class

### PR DESCRIPTION
**Pull request by ekoval**
Since "Nested" case requires some specific logic, it is moved to a separate class. Tested in UserService.

### CODE REVIEW

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

File: madprops/serializers.py [hunk 1]
```diff
@@ -170,6 +170,21 @@ def _add_parent_obj_field(self, data):
...
+    @property
+    def fields(self):
+        # We need this to avoid validation of foreign key field in case
+        # properties are created together with properties owner. In this case
+        # parent field value is not known on property validation stage, so
+        # validation fails.
+        fields = super(PropertySerializer, self).fields
+        for (name, value) in dict(fields).items():
+            if name == self.opts.parent_obj_field:
+                del fields[name]
```
Comment (RayeN) [inline madprops/serializers.py]: can we just `fields.pop(self.opts.parent_obj_field, None)`?

File: madprops/serializers.py [hunk 2]
```diff
@@ -170,6 +170,21 @@ def _add_parent_obj_field(self, data):
         return data
 
 
+class NestedPropertySerializer(PropertySerializer):
+    """Base class for Nested Property class."""
+    @property
+    def fields(self):
+        # We need this to avoid validation of foreign key field in case
```
Comment (RayeN) [inline madprops/serializers.py]: `to skip validation` ?

File: madprops/serializers.py [hunk 3]
```diff
@@ -170,6 +170,21 @@ def _add_parent_obj_field(self, data):
...
+        # properties are created together with properties owner. In this case
+        # parent field value is not known on property validation stage, so
+        # validation fails.
```
Comment (RayeN) [inline madprops/serializers.py]: `because during validation stage reference to parent isn't known.` ?

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Comment by RayeN** [comment]
:+1: Looks fine to me

**Comment by snitch** [comment]
:sparkles: No lint errors found. :sparkles:

**Pull request merged**